static/js/components/source/source_publish/SourcePublish.jsx

```diff
@@ -0,0 +1,284 @@
+import React, { useState, useEffect } from "react";
+import makeStyles from "@mui/styles/makeStyles";
+import Tooltip from "@mui/material/Tooltip";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import PropTypes from "prop-types";
+import { useDispatch, useSelector } from "react-redux";
```

Comment:
Import order to verify maybe? Usually all the MUI stuff comes after the react stuff.

Reply:
✅ Done.